src/libsystemd/sd-journal/journal-file.c

```diff
@@ -2091,6 +2091,12 @@ static int journal_file_append_entry_internal(
         assert(ts);
         assert(items || n_items == 0);
 
+        if (ts->realtime < le64toh(f->header->tail_entry_realtime))
+                return log_debug_errno(SYNTHETIC_ERRNO(ETXTBSY),
+                                       "Realtime timestamp %" PRIu64 " smaller than previous realtime "
+                                       "timestamp %" PRIu64 ", refusing entry.",
+                                       ts->realtime, le64toh(f->header->tail_entry_realtime));
+
```

Comment:
makes sense to add, but can you make up a new error code here? we already use ETXTBSY for journal files "from the future", but this is a bit different, it's internally non-consistent journal files.

Maybe EREMCHG? That's currently not used by the journal sources.

Reply:
note that this is half redundant, given write_to_journal() already does a similar check. would be good adding a comment there, to mention this.

(I think it makes sense to keep both logics in place, even if one is redundant, since they have a different focus: one looks at all log entries, the other one only to those hitting a specific file)